lib/kubernetes-deploy/kubeclient_builder.rb

```diff
@@ -44,10 +45,12 @@ def build_apps_v1beta1_kubeclient(context)
     end
 
     def _build_kubeclient(api_version:, context:, endpoint_path: nil)
-      config = GoogleFriendlyConfig.read(ENV.fetch("KUBECONFIG"))
-      unless config.contexts.include?(context)
-        raise ContextMissingError, context
-      end
+      # Find a context defined in kube conf files that matches the input context by name
+      friendly_configs = config_files.map { |f| GoogleFriendlyConfig.read(f) }
+      config = friendly_configs.each.lazy.find { |c| c.contexts.include?(context) }
```

Comment:
What does `friendly_configs.each.lazy.find` buy us vs `friendly_configs.find` directly? I must admit I'm not super familiar with `.lazy`, but it seems like `.each` will return `self` in this case, and `.find` is invoked immediately regardless, and already stops traversing the array once it finds a match.

Reply:
You are right, find/detect will return the first element found instead of loading all contexts.
lazy is to wrapper of the Enumerator returned by each to generate anther enumerator Enumerator::Lazy to load element on an as-needed basis. Here, lazy is redundant.